surveymonkey/api_surveymonkey.py

```diff
@@ -86,22 +98,40 @@ def get_surveys(self, **kwargs):
         """
         Returns a list of surveys owned or shared with the authenticated user.
         """
+        cache_key = "{}-{}-{}".format(SURVEY_MONKEY_API_TAG, "all_surveys", self.client_id)
+        all_surveys_data = cache.get(cache_key)
+
+        if self.surveymonkey_api_cache_duration == 0:
+            cache.delete(cache_key)
+        elif all_surveys_data:
+            return all_surveys_data
+
         url = "{}/{}".format(
             API_BASE,
             "v3/surveys",
         )
         response = self.__call_api_get(url, kwargs)
 
+        # TODO add support for next pages.
         if response.status_code == 200:
```

Comment:
```suggestion
         if response.status_code == 200 and self.surveymonkey_api_cache_duration:
             cache.set(cache_key, response.json(), self.surveymonkey_api_cache_duration)
             return response.json()
         elif response.status_code == 200:
             return response.json()
```

Reply:
We don't have the xblock context here. @andrey-canon